component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -156,23 +160,52 @@ private String validateIcon(final Icon annotation, final Collection<String> erro
 
         if (annotation.value() == Icon.IconType.CUSTOM) {
             final String icon = annotation.custom();
-            final Set<File> svgs =
-                    of(classes).map(it -> new File(it, "icons/" + icon + ".svg")).filter(File::exists).collect(toSet());
-            if (svgs.isEmpty()) {
-                log.error("No 'icons/" + icon + ".svg' found, this will run in degraded mode in Talend Cloud");
+            Set<File> svgs;
+            Set<File> pngs;
+            // legacy checks
+            if (configuration.isValidateLegacyIcons()) {
+                svgs = of(classes)
+                        .map(it -> new File(it, ICONS + icon + ".svg"))
+                        .collect(toSet());
+                pngs = Stream.of(classes)
+                        .map(it -> new File(it, ICONS + icon + "_icon32.png"))
```

Comment:
And here the path is relative (don't start from "/"). When we do the same for themed icons, the path starts from "/"
group comment $2.2

Reply:
but here we're using the File class wih parent and child parameters, not need to be absolute.

Reply:
Oh, I see now. Thank you